File: f6ac0d98-chair/recon-32-34/int-skeptic/bfs.mjs
import { readFileSync, existsSync, statSync } from 'node:fs';
import { dirname, resolve, join } from 'node:path';
const ROOT = process.argv[2];
const ENTRY = process.argv[3];
const targets = process.argv.slice(4);
const seen = new Map(); // file -> parent
const q = [];
function res(spec, from){
  if(!spec.startsWith('.')) return null;
  let p = resolve(dirname(from), spec);
  const cands=[p, p+'.js', p+'.jsx', join(p,'index.js'), join(p,'index.jsx')];
  for(const c of cands){ if(existsSync(c) && statSync(c).isFile()) return c; }
  return null;
}
const staticRe = /(?:^|\n)\s*(?:import\s[^;]*?from\s*|import\s*|export\s[^;]*?from\s*)['"]([^'"]+)['"]/g;
function edges(file){
  const src = readFileSync(file,'utf-8');
  // strip dynamic imports: import( ... )
  const out=[]; let m;
  staticRe.lastIndex=0;
  while((m=staticRe.exec(src))!==null){
    // ensure not preceded by '(' i.e. dynamic
    out.push(m[1]);
  }
  return out;
}
const entry = resolve(ROOT, ENTRY);
seen.set(entry, null); q.push(entry);
while(q.length){
  const f=q.shift();
  let specs; try{specs=edges(f);}catch(e){continue;}
  for(const s of specs){
    const r=res(s,f); if(!r) continue;
    if(seen.has(r)) continue;
    seen.set(r,f); q.push(r);
  }
}
console.log('reachable modules:', seen.size);
for(const t of targets){
  const hit=[...seen.keys()].filter(k=>k.endsWith('/'+t));
  if(!hit.length){ console.log('NOT REACHABLE:', t); continue; }
  for(const h of hit){
    const chain=[]; let c=h; while(c){ chain.push(c.replace(ROOT+'/','')); c=seen.get(c); }
    console.log('REACHABLE:', t, '\n   ', chain.reverse().join('\n  -> '));
  }
}

File: f6ac0d98-chair/recon-32-34/bfs.py
import os,re,sys,json
ROOT=sys.argv[1]; ENTRY=sys.argv[2]
imp=re.compile(r"""(?:from|import)\s*\(?\s*['"](\.[^'"]+)['"]""")
def resolve(base,spec):
    p=os.path.normpath(os.path.join(os.path.dirname(base),spec))
    for c in (p,p+'.js',p+'.jsx',os.path.join(p,'index.js'),os.path.join(p,'index.jsx')):
        if os.path.isfile(os.path.join(ROOT,c)): return c
    return None
seen=set(); stack=[ENTRY]
while stack:
    f=stack.pop()
    if f in seen: continue
    seen.add(f)
    try: t=open(os.path.join(ROOT,f),encoding='utf8').read()
    except: continue
    for m in imp.findall(t):
        r=resolve(f,m)
        if r and r not in seen: stack.append(r)
total=sum(os.path.getsize(os.path.join(ROOT,f)) for f in seen)
print(len(seen),"modules,",total,"src bytes")
for pat in sys.argv[3:]:
    hits=[f for f in sorted(seen) if pat in f]
    print(f"  {pat}: {len(hits)}", hits[:6])
